improver/cube_combiner.py

```diff
@@ -222,8 +223,10 @@ def _check_dimensions_match(
         Raises:
             ValueError: If dimension coordinates do not match
         """
-        ref_coords = cube_list[0].coords(dim_coords=True)
-        for cube in cube_list[1:]:
+        test_cube_list = iris.cube.CubeList(cube_list.copy())
```

Comment:
The doc-string probably now needs to specify that this check excludes the var_names of the coordinates.

Reply:
Doc-string updated